gradle/scripts/lib/common-dependencies.gradle

```
@@ -178,19 +176,8 @@ configure(projectsWithFlags('java')) {
         dependencies {
             configurations.configureEach { configuration ->
                 // Add to resolvable configurations
-                if (configuration.canBeResolved && !configuration.canBeConsumed) {
-                    add(configuration.name, platform(dependencyManagementProject))
-                }
-
-                // Find version overrides in dependency declaration configurations
                 if (!configuration.canBeResolved && !configuration.canBeConsumed) {
```

Question) I'm wondering if we want to target resolvable dependencies instead (since these are the configurations which will actually be applied)
```suggestion
                if (configuration.canBeResolved && !configuration.canBeConsumed) {
```

---

There are three types of configurations.
- Bucket - `implementation`, `api` ...
- Resolvable - `compileClasspath`, `runtimeClasspath`, ...
- Consumable - `apiElement`, ...

We directly added the dependencies to resolvable configurations. In particular, platform dependencies are added to each configuration here. Unfortunately, in Gradle 8, bucket styles are only allowed to add dependencies, and exceptions are raised if the dependencies are added to resolvable ones.

So I changed to add the platform dependencies to the buckets. It should be fine because they will be eventually inherited into resolvable configurations. All resolvable configurations defined in the Gradle Java plugin have at least one bucket configuration.
https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph

The only exception was `shadedTestRuntime` created by `gradle-scripts` that does not have a bucket parent. I added `shadedTestImplementation` as a bucket so that we inject the platform dependencies into it. As `shadedTestRuntime` extends `shadedTestImplementation`, all dependencies added to `shadedTestImplementation` will be visible to `shadedTestRuntime`.

---

I understood the concept while reviewing the PR, but I'm not sure why I left this review 😅  Sorry about that.
In short, I understood that we can't add dependencies to a resolvable configuration 🙇